### ISSUE TITLE
In correct error handling after sending a message

### ISSUE THREAD

**Issue by t-wynants** [GridProtectionAlliance/PTPSync#2]
In the majority of the cases, the error handling looks wrong.  The different "send" network functions are successful when they return a positive value.

```
ssize_t netSendEvent(Octet*,UInteger16,NetPath*);
ssize_t netSendGeneral(Octet*,UInteger16,NetPath*);
ssize_t netSendPeerGeneral(Octet*,UInteger16,NetPath*);
ssize_t netSendPeerEvent(Octet*,UInteger16,NetPath*);
```
Are all implemented with `sendto`.

`sendto`
_Return value_
If no error occurs, sendto returns the **total number of bytes sent**, which can be less than the number indicated by len. Otherwise, a value of SOCKET_ERROR (-1) is returned.

The protocol uses a different definition of an error (` ret != 0`) where the send functions are used.
Example: 

```
if (!netSendEvent(ptpClock->msgObuf,SYNC_LENGTH,&ptpClock->netPath)) {
        toState(PTP_FAULTY,rtOpts,ptpClock);
        DBGV("Sync message can't be sent -> FAULTY state \n");
    } else {
        DBGV("Sync MSG sent ! \n");
        ptpClock->sentSyncSequenceId++;	
    }
```
This results in logging like:
(ptpd debug) event ANNOUNCE_INTERVAL_TIMEOUT_EXPIRES
**(ptpd debug) error sending multi-cast general message**
**(ptpd debug) Announce MSG sent !**
(ptpd debug) catch_alarm: elapsed 1
(ptpd debug) timerUpdate: timer 2 expired
(ptpd debug) event SYNC_INTERVAL_TIMEOUT_EXPIRES
**(ptpd debug) error sending multi-cast event message**
**(ptpd debug) Sync MSG sent !**

but we never enter the FAULTY state and we never recover from this situation.

**Comment by ritchiecarroll** [comment]
Wow - thank you for your most helpful review. The code was ported from old fork, so I suspected there may be issues just give it's age, but it was the only one I could find that worked with Windows. The code deviated so much from the mainline that it was near impossible for a simplistic compare.

That said, you changes seem straight forward enough - so I'll get these on the list for correction.

Thanks!
Ritchie

**Comment by ritchiecarroll** [comment]
Fixed with PR #3
